ignite/services/doctor/doctor.go

```diff
@@ -0,0 +1,144 @@
+package doctor
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/gobuffalo/genny/v2"
+
+	chainconfig "github.com/ignite/cli/ignite/config/chain"
+	"github.com/ignite/cli/ignite/pkg/cliui/colors"
+	"github.com/ignite/cli/ignite/pkg/cliui/icons"
+	"github.com/ignite/cli/ignite/pkg/cosmosgen"
+	"github.com/ignite/cli/ignite/pkg/events"
+	"github.com/ignite/cli/ignite/pkg/gomodulepath"
+	"github.com/ignite/cli/ignite/templates/app"
+)
+
+// DONTCOVER: Doctor read and write the filesystem intensively, so it's better
+// to rely on integration tests only. See integration/doctor package.
+type Doctor struct {
```

Comment:
It could be an improvement in a different issue, but it might be a good idea to confirm with the user before the doctor makes any actual changes. User might be using Git which allows undoing changes though.

Reply:
For the first version of `doctor` I think this is fine, especially since the command is hidden.